java/client/src/main/java/glide/connectors/handlers/PubSubMessageQueue.java

```diff
@@ -0,0 +1,55 @@
+/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.connectors.handlers;
+
+import glide.api.models.PubSubMessage;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/** A FIFO message queue for {@link PubSubMessage}. */
+public class PubSubMessageQueue {
```

Comment:
Could be package-private.

Reply:
Compilation fails - `BaseClient` from `glide.api` uses it.

Reply:
That seems strange that it uses it directly. I would've thought it goes through MessageHandler.

Reply:
OK I see, you are creating a MessageHandler before the builder, then passing the handler in when creating the channel but passing the queue in to the client's constructor.

This is exposing a class from a non-public package to a non-private field in a public/protected class in an exposed package.

I am actually seeing that on a bunch of BaseClient classes. Let's discuss offline.

Reply:
Separate from the above issue of internal classes in public APIs, I think BaseClient should not have a MessageQueue. It should always route through a MessageHandler that accesses a MessageQueue. I would MessageQueue a static inner class of MessageHandler (package-private for unit testing).

This would make it clear a MessageQueue can't really be on its own and is always tied to a MessageHandler.